dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4396,6 +4396,54 @@ class FunctionRepeat : public IFunction
 };
 
 
+class FunctionSpace : public IFunction
+{
+public:
+    static constexpr auto name = "space";
+
+    FunctionSpace() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionSpace>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 1; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (arguments.size() != 1)
+            throw Exception(
+                fmt::format("Number of arguments for function {} doesn't match: passed {}, should be 1.", getName(), arguments.size()),
+                ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        const ColumnPtr c0_col = block.getByPosition(arguments[0]).column;
+
+        Field res_field;
+        int val_num = c0_col->size();
+        auto col_res = ColumnString::create();
+        col_res->reserve(val_num);
+
+        for (int i = 0; i < val_num; i++)
+        {
+            c0_col->get(i, res_field);
```

Comment:
Don't need to get res_field everytime, it's in-efficient, you can refer to `TidbPadImpl` as an example.

Reply:
Also need to some optimzations for constant column.